fvm/evm/emulator/emulator_test.go

```diff
@@ -0,0 +1,371 @@
+package emulator_test
+
+import (
+	"fmt"
+	"math"
+	"math/big"
+	"testing"
+
+	gethCommon "github.com/ethereum/go-ethereum/common"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	gethParams "github.com/ethereum/go-ethereum/params"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/fvm/evm/emulator"
+	"github.com/onflow/flow-go/fvm/evm/emulator/database"
+	"github.com/onflow/flow-go/fvm/evm/testutils"
+	"github.com/onflow/flow-go/fvm/evm/types"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+var blockNumber = big.NewInt(10)
+var defaultCtx = types.NewDefaultBlockContext(blockNumber.Uint64())
+
+func RunWithTestDB(t testing.TB, f func(types.Database)) {
+	testutils.RunWithTestBackend(t, func(backend types.Backend) {
+		testutils.RunWithTestFlowEVMRootAddress(t, backend, func(flowEVMRoot flow.Address) {
+			db, err := database.NewDatabase(backend, flowEVMRoot)
+			require.NoError(t, err)
+			f(db)
+		})
+	})
+}
+
+func RunWithNewEmulator(t testing.TB, db types.Database, f func(*emulator.Emulator)) {
+	env := emulator.NewEmulator(db)
+	f(env)
+}
+
+func RunWithNewBlockView(t testing.TB, em *emulator.Emulator, f func(blk types.BlockView)) {
+	blk, err := em.NewBlockView(defaultCtx)
+	require.NoError(t, err)
+	f(blk)
+}
+
+func RunWithNewReadOnlyBlockView(t testing.TB, em *emulator.Emulator, f func(blk types.ReadOnlyBlockView)) {
+	blk, err := em.NewReadOnlyBlockView(defaultCtx)
+	require.NoError(t, err)
+	f(blk)
+}
+
+func TestNativeTokenBridging(t *testing.T) {
+	RunWithTestDB(t, func(db types.Database) {
+		originalBalance := big.NewInt(10000)
+		testAccount := types.NewAddressFromString("test")
+
+		t.Run("mint tokens to the first account", func(t *testing.T) {
+			RunWithNewEmulator(t, db, func(env *emulator.Emulator) {
+				RunWithNewBlockView(t, env, func(blk types.BlockView) {
+					amount := big.NewInt(10000)
```

Comment:
if I understand the test correctly then it's safer to:
```suggestion
					amount := originalBalance
```

Reply:
fixed at [c848b92](https://github.com/onflow/flow-go/pull/4927/commits/c848b9212ca242dd68b095c4a3cd63058af8c377)